spec/connection_proxy_spec.rb

```diff
@@ -101,18 +93,6 @@
       @proxy.next_slave!
       @proxy.select_one(@sql)
     end
-
-    it 'should not switch when in a with_master-block' do
```

Comment:
@cainlevy whatcha think about a positive test: `should switch when in a master block`?